corda-common/src/main/java/ch/cordalo/corda/common/contracts/StateVerifier.java

```diff
@@ -618,25 +618,42 @@ protected StateVerifier verify() {
     }
     class MoreThanZero extends Size {
         protected MoreThanZero(StateVerifier parent) {  super(parent); }
-        protected MoreThanZero(StateVerifier parent, int size) {  super(parent, size); }
         @Override
         protected StateVerifier verify() {
-            super.verify();
             requireThat(req -> {
                 req.using(s("List must contain at least 1 entry."), this.getParentList().size() > 0);
                 return null;
             });
             return this;
         }
     }
-    class MoreThanOne extends Size {
-        protected MoreThanOne(StateVerifier parent) {  super(parent); }
-        protected MoreThanOne(StateVerifier parent, int size) {  super(parent, size); }
+    class MoreThanN extends Size {
+        protected MoreThanN(StateVerifier parent) {  this(parent, 1); }
+        protected MoreThanN(StateVerifier parent, int size) {  super(parent, size); }
+        @Override
+        protected StateVerifier verify() {
+            requireThat(req -> {
+                if (this.size() == 1) {
+                    req.using(s("List must have more than " + this.size() + " entry."), this.getParentList().size() > this.size());
+                } else {
+                    req.using(s("List must have more than " + this.size() + " entries."), this.getParentList().size() > this.size());
+                }
+                return null;
+            });
+            return this;
+        }
+    }
+    class LessThanN extends Size {
+        protected LessThanN(StateVerifier parent) {  this(parent, 1); }
+        protected LessThanN(StateVerifier parent, int size) {  super(parent, size); }
         @Override
         protected StateVerifier verify() {
-            super.verify();
             requireThat(req -> {
-                req.using(s("List must more than 1 entry."), this.getParentList().size() > 1);
+                if (this.size() == 1) {
+                    req.using(s("List must have less than " + this.size() + " entry."), this.getParentList().size() < this.size());
+                } else {
+                    req.using(s("List must have less than " + this.size() + " entries."), this.getParentList().size() < this.size());
+                }
```

Comment:
use instead (no if, less code)
`req.using(s("List must have less than " + this.size() + " entrie(s)."), this.getParentList().size() < this.size());`